src/Compilers/CSharp/Portable/Symbols/ExtendedErrorTypeSymbol.cs

```diff
@@ -31,14 +33,14 @@ internal ExtendedErrorTypeSymbol(CSharpCompilation compilation, string name, int
         {
         }
 
-        internal ExtendedErrorTypeSymbol(NamespaceOrTypeSymbol containingSymbol, string name, int arity, DiagnosticInfo errorInfo, bool unreported = false, bool variableUsedBeforeDeclaration = false)
+        internal ExtendedErrorTypeSymbol(NamespaceOrTypeSymbol? containingSymbol, string name, int arity, DiagnosticInfo? errorInfo, bool unreported = false, bool variableUsedBeforeDeclaration = false)
         {
-            Debug.Assert(((object)containingSymbol == null) ||
+            Debug.Assert(((object?)containingSymbol == null) ||
                 (containingSymbol.Kind == SymbolKind.Namespace) ||
                 (containingSymbol.Kind == SymbolKind.NamedType) ||
                 (containingSymbol.Kind == SymbolKind.ErrorType));
 
-            Debug.Assert(name != null);
+            RoslynDebug.Assert(name != null);
```

Comment:
>RoslynDebug [](start = 12, length = 11)

Was this change necessary?

Reply:
Yes. The netstandard version of `Debug` lacks the `[DoesNotReturnIf(false)]` attribute hence it would cause the compiler to believe that `name` was `null` going forward and invalidate the `_name = name` assignment.

---
In reply to: [364482752](https://github.com/dotnet/roslyn/pull/39192#discussion_r364482752) [](ancestors = 364482752)

Reply:
Ah, forgot about that. Asserting would introduce a maybe-null state :P